pkg/storage/s3/s3.go

```diff
@@ -330,39 +413,40 @@ func (d *driver) CreateStorage(cr *opapi.ImageRegistry, modified *bool) error {
 		util.UpdateCondition(cr, opapi.StorageIncompleteUploadCleanupEnabled, operatorapi.ConditionTrue, "Enable Cleanup Successful", "Default cleanup of incomplete multipart uploads after one (1) day was successfully enabled", modified)
 	}
 
-	cr.Status.Storage.State.S3 = d.Config
-	cr.Status.Storage.Managed = true
-
-	util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionTrue, "Creation Successful", "S3 bucket was successfully created", modified)
-
 	return nil
 }
 
 // RemoveStorage deletes the storage medium that we created
 func (d *driver) RemoveStorage(cr *opapi.ImageRegistry, modified *bool) error {
-	if !cr.Status.Storage.Managed {
-		return nil
-	} else if len(cr.Status.Storage.State.S3.Bucket) == 0 {
+	if !cr.Status.StorageManaged || len(d.Config.Bucket) == 0 {
 		return nil
 	}
 
-	svc, err := d.getSVC()
+	svc, err := d.getS3Service()
 	if err != nil {
 		return err
 	}
 	_, err = svc.DeleteBucket(&s3.DeleteBucketInput{
-		Bucket: aws.String(cr.Status.Storage.State.S3.Bucket),
+		Bucket: aws.String(d.Config.Bucket),
 	})
 
 	if err != nil {
 		if aerr, ok := err.(awserr.Error); ok {
 			util.UpdateCondition(cr, opapi.StorageExists, operatorapi.ConditionUnknown, aerr.Code(), aerr.Error(), modified)
 		}
+		return err
```

Comment:
What if we get timeout while reading the response from the server?

Reply:
Yes, i'll put that in the next pr.